### PR TITLE
Old style for classic block

### LINKED ISSUE #4926: Update Classic block to better resemble the current editor
The Classic block has caused confusion in a number of ways since the start, which has caused the name to change from "Freeform" to "Classic Text" to "Classic". It's also hard to visually distinguish from the Paragraph blocks, causing confusion because the interactions are very different. Alignments in the paragraph block work at the block level, whereas you can have multiple paragraphs with different alignments inside a Classic block, as well as a multitude of other content.

What if we made the Classic block reflect the styling of the current editor? It would have the following benefits:

- Reduce confusion as to the difference between a Paragraph block and the Classic block
- Provide some recognition for users migrating to Gutenberg
- By not changing the UI, we can better support TinyMCE specific plugins like TinyMCE advanced

Here's a mockup:

classic_editor_block

### PR DESCRIPTION
**Pull request by ellatrix**
## Description
Fixes #4926. Gives the classic block the old look.

One problem: the toolbar will only start showing on first time focus. :( @spocke, would it be possible at all to add a setting to the modern/inline theme to always show the toolbar? It might also be good to make a separate "theme" for this as we're not using quite a bit of logic in it.

## How Has This Been Tested?
Create a classic block.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code has proper inline documentation.

### CODE REVIEW

**Comment by mtias** [comment]
This is looking good.

Agreed on fixing the first time focus. It actually doesn't work unless you re-focus when you insert via auto-complete:

image

**Comment by mtias** [comment]
Related: #4928.

**Comment by iseulde** [comment]
I'll copy over parts of the "inline" theme from TinyMCE in the next commit so we have more control.

**Comment by iseulde** [comment]
I don't really see a way of fixing the focus problem without having to copy over a significant amount of code to the classic block to overwrite the theme. It would be nicer if the inline theme just had a setting to show toolbars all the time instead of just when the editor is focussed.

**Comment by iseulde** [comment]
Or for `createToolbars` etc. to be part of TinyMCE's UI module.

**Comment by jasmussen** [comment]
Sorry I missed this. This is looking really good. 

Here's the toolbar issue in GIF form:

classic

It would be _nice_ if we could address this, but it doesn't feel like the end of the world to me. Unless we can do some simple hack like run an onload focus for each classic block (hacky), I think it might be worth getting this in and iterating.

**Comment by jasmussen** [comment]
Am seeing this error, though:

screen shot 2018-03-07 at 10 47 58

Possibly unrelated?

**Comment by iseulde** [comment]
@jasmussen I don't like the idea of manually moving around focus when the page initialises, but it could work... cc @afercia

**Comment by jasmussen** [comment]
I can live without it, it'd just be nice if we could find a way to show the toolbar on the classic blocks from the start. But can live without it.

**Comment by iseulde** [comment]
Rebased, hopefully the error is gone.

Also added the focus and attempted to reset focus to the previous element, but for some reason it still has the caret in the last classic block... TBH I can live with this so the UI shows up while we wait for a setting in TinyMCE.

**Comment by iseulde** [comment]
Cc @spocke or @Afraithe, would it be possible to add a setting to the inline mode of the "modern" TinyMCE theme to always show the toolbar (as opposed to show/hide on focus/blur)?

**Comment by spocke** [comment]
What we could do is to add a new way of initializing the editor. Currently there is inline and iframe mode what's needed here is some form of inplace editing where a div is converted to contenteditable and the ui is appended before that target element. Not exactly sure how that should be exposed though the api though. But this is something we have seen people asking for before being able to convert div elements to editors but have the ui in place and not hide on blur.

**Comment by jasmussen** [comment]
I feel like maybe we should merge this in as is, and then open a separate ticket based on https://github.com/WordPress/gutenberg/pull/4948#issuecomment-372258376. Thoughts, @gziolo @karmatosed ?

**Comment by karmatosed** [comment]
> I feel like maybe we should merge this in as is, and then open a separate ticket based on #4948 (comment). Thoughts, @gziolo @karmatosed ?

Seems like a very solid plan @jasmussen.

**Comment by gziolo** [comment]
Have you tested it with more than one instance of the classic editor?
Have you tested it with the classic editor as a block in the middle of the post?
The fact that we set focus on init might bring unexpected behavior. I can test it quickly myself, too.

**Comment by gziolo** [comment]
As I expected, the focus stays in the last Classic block in the document after it gets loaded. I don't think this is an expected behavior. We should look into an alternative solution to make this work.

**Comment by iseulde** [comment]
@gziolo See the PR description and comments. No easy solution. Maybe we should just keep on hiding it for now and only update the styles.

**Comment by gziolo** [comment]
Another idea, after checking the screencast from @jasmussen. Would it help to add a placeholder for the toolbar to avoid this flickering effect?

**Comment by gziolo** [comment]
By the way, can we add an icon to convert to blocks inside the toolbar? That would be awesome 😍

**Comment by jasmussen** [comment]
Both of those are good ideas. We have a hover label currently, perhaps that could be the "until you select" toolbar.

**Comment by iseulde** [comment]
@jasmussen @gziolo Updated. This should be a lot better. Still not perfect as long as we can't create the toolbar immediately.

**Comment by iseulde** [comment]
> By the way, can we add an icon to convert to blocks inside the toolbar? That would be awesome 😍

Sure, but separately please :)

**Comment by jasmussen** [comment]
Nice work Ella, here's a GIF:

classic

I pushed a little polish so it looks like this:

classic now

That included removing the hover label. This label is likely to go through some iteration regardless, and perhaps become a drag handle, but once we do that it migh sit to the left instead of the top.

**Comment by iseulde** [comment]
@jasmussen @gziolo Thanks for the changes and testing!

**Comment by gziolo** [comment]
Agreed with @jasmussen, this looks great and make it very clear that we are dealing with Classic block. Thanks for making all the adjustments to this PR, excellent job @iseulde 💯

**Review by gziolo — APPROVED** [review]
@jasmussen is fixing one more small UI glitch and we should merge 👍

File: blocks/library/freeform/editor.scss [hunk 1]
```diff
@@ -140,7 +140,7 @@ div[data-type="core/freeform"] .editor-block-contextual-toolbar + div {
 
 .freeform-toolbar {
 	width: auto;
-	margin: - $block-padding;
+	margin: -#{ $block-padding + 1 } -#{ $block-padding } -#{ $block-padding } -#{ $block-padding };
```
Comment (iseulde) [inline blocks/library/freeform/editor.scss]: I think this code deserves a comment. :)

Comment (jasmussen) [reply blocks/library/freeform/editor.scss]: 😂 

On it in a minute.

Comment (iseulde) [reply blocks/library/freeform/editor.scss]: I also lost the top border now.

screen shot 2018-04-13 at 10 51 43

Comment (gziolo) [reply blocks/library/freeform/editor.scss]: It works, computes px value :)

**Comment by iseulde** [comment]
Okay :) We can iterate in separate PRs :)

**Comment by jasmussen** [comment]
🔥 🔥 🔥 🔥

**Pull request merged**